test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresWithLambdas.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Security.Permissions;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Helpers;
+
+namespace Mono.Linker.Tests.Cases.RequiresCapability
+{
+	[SkipKeptItemsValidation]
+	public class RequiresWithLambdas
+	{
+		public static void Main ()
+		{
+			typeof (RequiresOnClass).RequiresAll ();
+			typeof (NoRequiresOnClass).RequiresAll ();
+		}
+
+		[RequiresUnreferencedCode (nameof (RequiresOnClass))]
+		public class RequiresOnClass
+		{
+			public delegate void action ();
+			public void RucMethod () { }
+			public void InvokeLocalRucAction ()
+			{
+				Action x = () => { RucMethod (); };
+				x ();
+			}
+
+			public Action ReturnRucAction ()
+			{
+				return () => { RucMethod (); return; };
+			}
+
+			public void InvokeReturnedRucAction ()
+			{
+				var f = ReturnRucAction ();
+				f ();
+			}
+
+			public void InvokeActionArg (action f)
+			{
+				f ();
+			}
+
+			public void PassRucMethodToBeInvoked ()
+			{
+				InvokeActionArg (RucMethod);
+			}
+		}
+
+		public class NoRequiresOnClass
+		{
+			public delegate void action ();
+
+			[RequiresUnreferencedCode (nameof (RucMethod))]
+			public void RucMethod () { }
+
+			[ExpectedWarning ("IL2026", nameof (RucMethod), ProducedBy = ProducedBy.Analyzer)]
+			public void InvokeLocalRucAction ()
+			{
+				Action x = () => { RucMethod (); };
+				x ();
+			}
+
+			// Bug - Shouldn't warn since annotated with RUC
+			[ExpectedWarning ("IL2026", ProducedBy = ProducedBy.Analyzer)]
+			[RequiresUnreferencedCode (nameof (ReturnRucAction))]
+			public Action ReturnRucAction ()
+			{
+				return () => { RucMethod (); return; };
+			}
+
+			// Bug - Should warn in Trimmer
+			[ExpectedWarning ("IL2026", nameof (ReturnRucAction), ProducedBy = ProducedBy.Analyzer)]
+			public void InvokeReturnedRucAction ()
+			{
+				var f = ReturnRucAction ();
+				f ();
+			}
```

Comment:
This is really weird - I do agree that on the surface this should warn. I'll have to look into it.

Reply:
I'm not sure why, but as I fixed other parts of the tests, this warning started appearing in the linker.